File: 2018/tests/Day20Test.hs
module Day20Test where

import           Day20
import           Test.Tasty.Hspec

spec_Day20 = do

  describe "Part 1" $ do
    it "example 1" $ part1 "^WNE$" `shouldBe` 3
    it "example 2" $ part1 "^ENWWW(NEEE|SSE(EE|N))$" `shouldBe` 10
    it "example 3" $ part1 "^ENNWSWW(NEWS|)SSSEEN(WNSE|)EE(SWEN|)NNN$" `shouldBe` 18
    it "example 4" $ part1 "^ESSWWN(E|NNENN(EESS(WNSE|)SSS|WWWSSSSE(SW|NNNE)))$" `shouldBe` 23
    it "example 5" $ part1 "^WSSEESWWWNW(S|NENNEEEENN(ESSSSW(NWSW|SSEN)|WSWWN(E|WWS(E|SS))))$" `shouldBe` 31
    it "my input" $ (part1 <$> readInput) >>= (`shouldBe` 3966)

  describe "Part 2" $ do
    it "my input" $ (part2 <$> readInput) >>= (`shouldBe` 8173)

readInput = head . lines <$> readFile "data/day20.in"

File: 2022/test/day20_test.clj
(ns day20-test
  (:require
   [clojure.java.io :as io]
   [clojure.test :refer [deftest is testing]]
   [day20 :refer [part1 part2]]))

(def my-input (slurp (io/resource "input20.txt")))
(def example "1 2 -3 3 -2 0 4")

(deftest part1-test
  (testing "example"
    (is (= 3 (part1 example))))
  (testing "my input"
    (is (= 11616 (part1 my-input)))))

(deftest part2-test
  (testing "example"
    (is (= 1623178306 (part2 example))))
  (testing "my input"
    (is (= 9937909178485 (part2 my-input)))))
